girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -106,6 +107,7 @@ def __init__(self, assetstore):
         # happens to existing assetstores that no longer can access their temp
         # directories.
         self.tempDir = os.path.join(self.assetstore['root'], 'temp')
+        self.deleteLock = threading.Lock()
```

Comment:
A new assestore adapter instance is created in each request thread, but it looks like this lock only protects against concurrent access on the same adapter instance.

My understanding of the race condition was that it would only happen across multiple request threads, is that not the case?